### PR TITLE
[Google Map] Correctly check map configurations on Widget update

### PR DESCRIPTION
**Pull request by KlausJokinen**
## Description

*Correctly check map configurations on Widget update*
 *  _updateOptions();
 *  _updateMarkers();
 *  _updatePolygons();
 *  _updatePolylines();
 *  _updateCircles();

Current implementation didn't check for changes*  in the above configurations causing an error when sending incorrect values to methodChannel.

* `_updateMapOptions `didn't check for List equality on `updatesMap `method when doing a distinct check.

## Related Issues

*flutter/flutter#47619.*

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [X] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by antonbt** [comment]
This is much needed. Please approve this PR.

File: packages/google_maps_flutter/lib/src/google_map.dart [hunk 1]
```diff
@@ -253,6 +253,7 @@ class _GoogleMapState extends State<GoogleMap> {
 
   @override
   void didUpdateWidget(GoogleMap oldWidget) {
+    print('LOCAL');
```
Comment (theochampion) [inline packages/google_maps_flutter/lib/src/google_map.dart]: You may want to remove debug log ;)

Comment (KlausJokinen) [reply packages/google_maps_flutter/lib/src/google_map.dart]: It should be already removed 😁

**Comment by KlausJokinen** [comment]
@iskakaushik Any progress regarding the PR?

File: AUTHORS [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Below is a list of people and organizations that have contributed
+﻿# Below is a list of people and organizations that have contributed
```
Comment (cyanglaz) [inline AUTHORS:1]: hmm, not sure what the diff is here. Please revert.

File: packages/google_maps_flutter/lib/src/google_map.dart [hunk 3]
```diff
@@ -265,6 +265,7 @@ class _GoogleMapState extends State<GoogleMap> {
     final _GoogleMapOptions newOptions = _GoogleMapOptions.fromWidget(widget);
     final Map<String, dynamic> updates =
         _googleMapOptions.updatesMap(newOptions);
+
```
Comment (cyanglaz) [inline packages/google_maps_flutter/lib/src/google_map.dart:268]: plz revert

File: packages/google_maps_flutter/lib/src/google_map.dart [hunk 4]
```diff
@@ -312,7 +341,9 @@ class _GoogleMapState extends State<GoogleMap> {
       widget.initialCameraPosition,
       this,
     );
+
```
Comment (cyanglaz) [inline packages/google_maps_flutter/lib/src/google_map.dart:344]: please revert, ditto below

**Comment by KlausJokinen** [comment]
Ohh this PR is still open, I need to check asap what is the current status of map configurations and do these modifications still apply.

**Comment by cyanglaz** [comment]
@KlausJokinen Greetings :) Wanted to check in to see if you are still working on this PR.

**Comment by KlausJokinen** [comment]
@cyanglaz Hi, It's been quite long time since I opened this PR and lot of have changed since then. I think this PR can be closed and I will re-open new one when I have more time 👍

**Comment by cyanglaz** [comment]
@KlausJokinen Sounds good! Will close the PR. Please feel free to reopen and ping me :)

**Pull request closed without merging**